### PR TITLE
BandPower - Fixes #44, #45, and #46

### LINKED ISSUE #44: Request: sigproc Spectrogram Collection
I think it's a pretty common use-case to put a Window in front of Spectrum. This should probably be in a pre-made collection with easy-access settings.

(1-of-several requests to get at SpectralBandpower)

### PR DESCRIPTION
**Pull request by cboulay**
I don't expect this to be merged without major revisioin, but I hope it's a good starting point for discussing adding BandPower to sigproc.

### CODE REVIEW

**Comment by cboulay** [comment]
Yeah I still want to work it out a bit. Using the revised Window will help. I've had many more hours with ezmsg since I opened this PR and I'm confident I can make a cleaner implementation, or at least one that's more consistent in its naming conventions and parameterization.

File: src/ezmsg/core/message.py [hunk 1]
```diff
@@ -36,7 +36,13 @@ def __init__(self):
         )
 
 
-class Flag(Message):
+@dataclass
```
Comment (griffinmilsap) [inline src/ezmsg/core/message.py:39]: Good catch

**Comment by griffinmilsap** [comment]
this is massive.  it looks to me like this affects synth, window, axisarray, and introduces a few new features all together.  The feature set it adds looks REALLY nice to have, but there's only so much time in the day.  I'm going to review this by merging it into dev, running some of my existing software, and seeing if anything breaks; but just as a heads up I don't have the bandwidth to review every line of this PR and expect we'd have some followup fixes as necessary.

**Comment by cboulay** [comment]
Yes of course, I'm available to make updates as issues arise.

**Pull request merged**